### ISSUE TITLE
Bug in MissionRawImpl::set_current_mission_item_async()

### ISSUE THREAD

**Issue by steveb2014** [mavlink/MAVSDK#1936]
Found this check in the `MissionRawImpl::set_current_mission_item_async` method. The `&&` should be an `||` and should be an easy fix.

```cpp
if (index < 0 && index >= _mission_progress.last.total) {
        _parent->call_user_callback([callback]() {
            if (callback) {
                callback(MissionRaw::Result::InvalidArgument);
                return;
            }
        });
    }
```

**Comment by JonasVautherin** [comment]
Do you feel like opening a PR for it, so that you get the credit? :innocent:

**Comment by steveb2014** [comment]
Wow.  That's a great idea.  Never done that on a public project before.  
Will have to investigate!

**Comment by JonasVautherin** [comment]
Just ping me here if you need help or decide to give up. But I think it's worth it. It's something like this:

1. Fork the repo
2. Clone your forked repo (that will be your `origin` remote)
3. Create a new branch for your fix, say `git branch fix-mission-set-current-item`
4. Commit your changes, and push to github. In the terminal, you will see instructions for opening a PR (basically a link to open in your browser). If you miss that and open your repo in github, the UI will offer you to create a PR.
5. Create a PR that points to mavlink/MAVSDK. Bonus: in the PR description, write "Resolves #1936".
6. That's it, we should see it here!

**Comment by julianoes** [comment]
Whoops, good catch @steveb2014.
